aws/amazon-chime-sdk-js
# Fix uncaught promise exception for bindAudioOutput API

**Issue #:** https://github.com/aws/amazon-chime-sdk-js/issues/907

**Description of changes:** Checking for `browserBehavior.supportsSetSinkId()` before calling `AsyncScheduler() -> this.bindAudioOutput();` in [`DefaultDeviceController`](https://github.com/aws/amazon-chime-sdk-js/blob/59bf60c30c8450451d9d1e36ec30a5b67e787607/src/devicecontroller/DefaultDeviceController.ts#L402)

**Testing**

1. Have you successfully run `npm run build:release` locally?
Yes.

2. How did you test these changes?
Tested by running the `serverless demo` & the `meeting readiness checker` demo in Firefox & Safari browser to check if there is any issue while
    - Starting/Joining a meeting.
    - Choosing audio input device, mute, unmute operation.
    - Enable/Disable speaker from the meeting canvas.
    - Set the `media.setsinkid.enabled` preference to `true` in Firefox. This enables the use of `SetSinkId` API as mentioned here https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/setSinkId and verified this error does not show up in the browser console.

3. Can these changes be tested using the demo application? If yes, which demo application can be used to test it?
Yes. Serverless demo.

4. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved?
NA

5. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved?
NA

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
